js_modules/dagit/packages/core/src/instance/InstanceTabs.tsx

```diff
@@ -21,7 +21,9 @@ export const InstanceTabs: React.FC<Props> = (props) => {
         <Tab id="schedules" title="Schedules" to="/instance/schedules" />
         <Tab id="sensors" title="Sensors" to="/instance/sensors" />
         <Tab id="backfills" title="Backfills" to="/instance/backfills" />
-        <Tab id="config" title="Configuration" to="/instance/config" />
+        {queryData?.data?.instance.showInstanceConfig ? (
+          <Tab id="config" title="Configuration" to="/instance/config" />
+        ) : null}
```

Comment:
this does not work the way i thought it would, I get a "Tabs must render Tab instances" error when its not set. Can i get an assist from somebody who actually knows TSX :)

Reply:
I think this has to do with the tabs component which makes some assumptions around having non-null children.  I made the following change:

```
 export const Tabs = styled(({selectedTabId, children, onChange, size = 'large', ...rest}) => {
-  if (!(children instanceof Array) || children.some((c) => c.type !== Tab)) {
+  if (!(children instanceof Array) || children.filter((_) => !!_).some((c) => c.type !== Tab)) {
     throw new Error('Tabs must render Tab instances');
   }

   return (
     <div {...rest} role="tablist">
       {React.Children.map(children, (child) =>
-        React.cloneElement(child, {
-          selected: child.props.selected || child.props.id === selectedTabId,
-          $size: size,
-          ...(onChange
-            ? {
-                onClick: () => onChange(child.props.id),
-              }
-            : {}),
-        }),
+        child
+          ? React.cloneElement(child, {
+              selected: child.props.selected || child.props.id === selectedTabId,
+              $size: size,
+              ...(onChange
+                ? {
+                    onClick: () => onChange(child.props.id),
+                  }
+                : {}),
+            })
+          : null,
       )}
     </div>
   );
```